src/resmom/linux/alps.c

```diff
@@ -6929,15 +6930,27 @@ create_vnodes_KNL(basil_response_query_system_t *sys_knl)
 		for (node_idx = 0; node_idx < node_count; node_idx++) {
 
 			node_id = nid_arr[node_idx];
-
 			snprintf(vname, VNODE_NAME_LEN, "%s_%d", mpphost_knl, node_id);
 
+			if (first_compute_node) {
+				/*
+				 * Create the name of the very first vnode so we
+				 * can attach topology info to it.
+				 */
+
+				attr = ATTR_NODE_TopologyInfo;
+				if (vn_addvnr(vnlp, vname, attr, (char *) basil_inventory,
+					     ATR_TYPE_STR, READ_ONLY, NULL) == -1)
+					  goto bad_vnl;
+				first_compute_node = 0;
```

Comment:
@dilip-krishnan  I think we should also set this variable to 1 when mom reads inventory again (maybe when HUP is sent to it). Because the second time it will just not enter this part of the code.